script/lazytox.py

```diff
@@ -0,0 +1,204 @@
+        stdout, _ = await asyncio.gather(
+            read_stream(proc.stdout, sys.stdout.write),
+            read_stream(proc.stderr, sys.stderr.write))
+    exit_code = await proc.wait()
+    # Convert to ASCII
+    stdout = stdout.decode('utf-8')
+    # log = RE_ASCII.sub('', log.decode())
+    return exit_code, stdout
+
+
+async def git():
+    """Exec git."""
+    try:
+        with open('lazytox.log') as file:
+            oldfiles = file.readlines()
+    except FileNotFoundError:
+        oldfiles = []
+
+    try:
+        _, log = await async_exec(
+            'git', 'diff', 'upstream/dev...', '--name-only')
+    except FileNotFoundError:
+        print("Using a cached version of changed files.")
+        return '\n'.join(oldfiles)
+
+    if oldfiles != log:
+        with open('lazytox.log', 'w') as file:
+            file.write(log)
+    return log
+
+
+async def pylint(files):
+    """Exec pylint."""
+    try:
+        # Drops STDERR, it contains info on loading config file
+        _, log = await async_exec('pylint', '-f', 'parseable', *files,
+                                  mute_err=True)
+    except FileNotFoundError:
+        return []
+
+    res = []
+    for line in log.splitlines():
+        line = line.split(':')
+        if len(line) != 4:
+            continue
+        res.append(ERROR(line[0].replace('\\', '/'), line[1], "", line[2]))
+    return res
+
+
+async def flake8(files):
+    """Exec flake8."""
+    try:
+        _, log = await async_exec('flake8', *files)
+    except FileNotFoundError:
+        return []
+    res = []
+    for line in log.splitlines():
+        line = line.split(':')
+        if len(line) != 4:
+            continue
+        res.append(ERROR(line[0], line[1], line[2], line[3].strip()))
+    return res
+
+
+async def main():
+    """Main loop."""
+    # Ensure we are in the homeassistant root
+    os.chdir(os.path.dirname(os.path.dirname(os.path.realpath(__file__))))
+
+    files = await git()
+    if not files:
+        print("No changed files found. Please ensure you have added your "
+              "changes with git add & git commit")
+        return
+
+    files = files.splitlines()
+    pyfile = re.compile(r".+\.py$")
+    pyfiles = [file for file in files if pyfile.match(file)]
+
+    print("CHANGED FILES:", ' '.join(files))
+    print("============================")
+
+    fres, pres = await asyncio.gather(flake8(pyfiles), pylint(pyfiles))
+    res = fres + pres
+
+    res.sort(key=lambda item: item.file)
+
+    gen_req = False
+    lint_ok = True
+    test_files = set()
+
+    for err in res:
+        print("{} {}:{} {}".format(err.file, err.line, err.col, err.msg))
+
+        # Will run generate_requirements if components and lint passes
+        if err.file.startswith('homeassistant/components/'):
+            gen_req = True
+
+        # Ignore tests/ for the lint_ok test, but otherwise we have an issue
+        if not err.file.startswith('tests/'):
```

Comment:
`flake8` also runs over the `tests/` directoy AFAIK, so checking whether the file is inside the `tests/` dir isn't good enough to determine whether it's an actual lint issue.